### PR TITLE
feat(iac): cognito user pool and user groups

### PR DESCRIPTION
**Pull request by irumvanselme**

### CODE REVIEW

File: iac/auth/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    temporaryPasswordValidityDays: 7
+  },
+  schemas: [
+    {
+      attributeDataType: "String",
+      name: "email",
+      required: true,
+      mutable: true,  // Set to true if you want to allow users to change their email
+      stringAttributeConstraints: {
+        minLength: "0",
```
Comment (rav3n11) [inline iac/auth/src/index.ts:51]: a 0 length email is allowed?

File: iac/certificate/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const us_east_1 = new aws.Provider("us-east-1", {region: "us-east-1"});
+
+const _cert = new aws.acm.Certificate("certificate", {
+  domainName: hostedZone.domainName,
+  subjectAlternativeNames: [pulumi.interpolate`*.${hostedZone.domainName}`],
+  tags: {
+    Environment: "dev",
+  },
+  validationMethod: "DNS",
+}, {dependsOn: [], provider: us_east_1});
```
Comment (rav3n11) [inline iac/certificate/src/index.ts:23]: why us-east-1 here?

**Review by rav3n11 — APPROVED** [review]
all looks good otherwise

**Review by Fidesnoella — APPROVED** [review]

File: iac/auth/src/index.ts (hunk 1 above)
Comment (irumvanselme) [reply iac/auth/src/index.ts]: Fixed

**Pull request merged**